src/game_updater/GameUpdater.cpp

```diff
@@ -167,28 +168,58 @@ void GameUpdater::UpdatePlayer(std::shared_ptr<GameState> game_state) {
   }
 }
 
+std::shared_ptr<Platform> GetCurrentPlatform(std::shared_ptr<Level> level) {
+  return level->getPlatforms()->at(std::max(
+      0l,
+      std::min((long)(level->getMusic()->getPlayingOffset().asMilliseconds() /
+                      (float)MS_PER_PLATFORM),
+               (long)level->getPlatforms()->size() - 1)));
+}
```

Comment:
If we're doing that this shouldn't be merged at all, so up to you guys. I think it looks better to follow the platforms but the logic could get more complicated as you mentioned.